lib/input.test.js

```diff
@@ -0,0 +1,59 @@
+const { isValid, parse } = require('./input');
+const { add, execute } = require('./notes');
+
+
+describe('validate the commands correct', () => {
+    it('validates', () => {
+        const obj = {
+            type: 'add',
+            payload: 'A note'
+        };
+
+        const result = isValid(obj);
+        
+        expect(result).toBeTruthy();
+    });
+});
+
+describe('parses the commandline into an object', () => {
+    it('parses array into object?', () => {
+        const array = ['node', 'index.js', '--add', 'My Note'];
+
+        const result = parse(array);
+
+        expect(result).toEqual({
+            type: 'add',
+            payload: 'My Note'
+        });
+    });
+});
+
+describe('create an object representing a note', () => {
```

Comment:
these last two tests should go into their own file `note.test.js`

Reply:
Completed with class refactoring